src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/ClientConfigurationLanguageSettings.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.lsp4ij.server.definition.launching;
+
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.Map;
+
+/**
+ * Language-specific client-side settings for a user-defined language server configuration.
+ */
+public class ClientConfigurationLanguageSettings {
```

Comment:
These are the actual per-language client-side settings. Again, there's only one right now, `caseSensitive`, but it I think we've discussed a few others in the context of other issues/PRs.